root/etc/cont-init.d/01-migrations

```diff
@@ -1,30 +1,33 @@
-#!/bin/bash
+#!/usr/bin/with-contenv bash
+# shellcheck shell=bash
 
 MIGRATIONS_DIR="/migrations"
 MIGRATIONS_HISTORY="/config/.migrations"
 
 echo "[migrations] started"
 
 if [[ ! -d ${MIGRATIONS_DIR} ]]; then
-  echo "[migrations] no migrations found"
-  exit
+    echo "[migrations] no migrations found"
+    exit
 fi
 
 for MIGRATION in $(find ${MIGRATIONS_DIR}/* | sort -n); do
-  NAME="$(basename "${MIGRATION}")"
-  if [[ -f ${MIGRATIONS_HISTORY} ]] && grep -Fxq "${NAME}" ${MIGRATIONS_HISTORY}; then
-    echo "[migrations] ${NAME}: skipped"
-    continue
-  fi
-  echo "[migrations] ${NAME}: executing..."
-  chmod +x "${MIGRATION}"
-  EXIT_CODE=$(/bin/bash "${MIGRATION}"; echo $?)
-  if [[ ${EXIT_CODE} -ne 0 ]]; then
-    echo "[migrations] ${NAME}: failed with exit code ${EXIT_CODE}, contact support"
-    exit "${EXIT_CODE}"
-  fi
-  echo "${NAME}" >> ${MIGRATIONS_HISTORY}
-  echo "[migrations] ${NAME}: succeeded"
+    NAME="$(basename "${MIGRATION}")"
+    if [[ -f ${MIGRATIONS_HISTORY} ]] && grep -Fxq "${NAME}" ${MIGRATIONS_HISTORY}; then
+        echo "[migrations] ${NAME}: skipped"
+        continue
+    fi
+    echo "[migrations] ${NAME}: executing..."
+    chmod +x "${MIGRATION}"
+    # Execute migration script in a subshell to prevent it from modifying the current environment
+    ("${MIGRATION}")
+    EXIT_CODE=$?
```

Comment:
This is the only significant change. Don't run the migration inside a subshell that gets set as the value of the `EXIT_CODE` variable! If the migration script has any output (echo, or anything) it will be included in the value of the variable. the variable should ONLY include the exit code. Still run the migration in a subshell (in parentheses), but do that separately from setting the `EXIT_CODE` variable.